go/vt/discovery/keyspace_events.go

```diff
@@ -401,7 +556,10 @@ func newKeyspaceState(kew *KeyspaceEventWatcher, cell, keyspace string) *keyspac
 		keyspace: keyspace,
 		shards:   make(map[string]*shardState),
 	}
-	kew.ts.WatchSrvKeyspace(context.Background(), cell, keyspace, kss.onSrvKeyspace)
+	kew.ts.WatchSrvKeyspace(kew.ctx, cell, keyspace, kss.onSrvKeyspace)
+	if !GetHackVTGateTestMode() {
+		kew.ts.WatchSrvVSchema(kew.ctx, cell, kss.onSrvVSchema)
+	}
 	return kss
```

Comment:
as discussed offline, this can be subscribed to existing watcher in vschema_manager.